Add a "pt2 compliant" tag; add config to graph break on non-pt2_compliant ops

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #112053
* __->__ #111933
* #111948
* #111915
* #111912

This PR:
- adds the pt2 compliant tag. This tag specifies that the operator works
  with the PT2 compilation APIs. A custom op author should test their
  ops with opcheck if they choose to add this tag.
- adds a config for Dynamo to allow only pt2 compliant ops into the
  graph and graph break on all other OpOverload/OpOverloadPacket.

Bikeshedding help wanted on the name of the tag. It should be easily
grep-able so we can set up rules for it.

Test Plan:
- new tests

Reviewers:

Subscribers:

Tasks:

Tags:

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

zou3519 committed:
Add pt2_compliant tag; add config to graph break on non-pt2_compliant ops

This PR:
- adds the pt2_compliant tag. This specifies that the operator works
  with the PT2 compilation APIs. A custom op author should test their
  ops with opcheck if they choose to add this tag.
- adds a config for Dynamo to allow only pt2_compliant ops into the
  graph and graph break on all other OpOverload/OpOverloadPacket.

Bikeshedding help wanted on the name of the tag. It should be easily
grep-able so we can set up rules for it.

Test Plan:
- new tests

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

zou3519 committed:
Update on "Add pt2_compliant tag; add config to graph break on non-pt2_compliant ops"


This PR:
- adds the pt2_compliant tag. This specifies that the operator works
  with the PT2 compilation APIs. A custom op author should test their
  ops with opcheck if they choose to add this tag.
- adds a config for Dynamo to allow only pt2_compliant ops into the
  graph and graph break on all other OpOverload/OpOverloadPacket.

Bikeshedding help wanted on the name of the tag. It should be easily
grep-able so we can set up rules for it.

Test Plan:
- new tests

Reviewers:

Subscribers:

Tasks:

Tags:

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

zou3519 commented on aten/src/ATen/native/tags.yaml:
```diff
@@ -3,6 +3,15 @@
 - tag: inplace_view
   desc: |
           This tag indicates if an operator *only* modifies the tensor metadata
+- tag: pt2_compliant_tag
```
torch.Tag.pt2_compliant_tag and at::Tag::pt2_compliant_tag may sound a bit goofy, but they are easily greppable

zou3519 committed:
Update on "Add a "pt2 compliant" tag; add config to graph break on non-pt2_compliant ops"


This PR:
- adds the pt2 compliant tag. This tag specifies that the operator works
  with the PT2 compilation APIs. A custom op author should test their
  ops with opcheck if they choose to add this tag.
- adds a config for Dynamo to allow only pt2 compliant ops into the
  graph and graph break on all other OpOverload/OpOverloadPacket.

Bikeshedding help wanted on the name of the tag. It should be easily
grep-able so we can set up rules for it.

Test Plan:
- new tests

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

ezyang commented:
Bikeshed is fine.

Are you going to grandfather aten ops into this, or treat the aten namespace specially?

zou3519 commented:
> Bikeshed is fine.
> 
> Are you going to grandfather aten ops into this, or treat the aten namespace specially?

Plan is to grandfather all aten ops created via codegen into this by automatically adding the tag during codegen

zou3519 committed:
Update on "Add a "pt2 compliant" tag; add config to graph break on non-pt2_compliant ops"


This PR:
- adds the pt2 compliant tag. This tag specifies that the operator works
  with the PT2 compilation APIs. A custom op author should test their
  ops with opcheck if they choose to add this tag.
- adds a config for Dynamo to allow only pt2 compliant ops into the
  graph and graph break on all other OpOverload/OpOverloadPacket.

Bikeshedding help wanted on the name of the tag. It should be easily
grep-able so we can set up rules for it.

Test Plan:
- new tests

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

ezyang commented:
no, it's not necessary

zou3519 commented:
@pytorchbot merge

zou3519 commented:
@pytorchbot merge

yanboliang commented on torch/_dynamo/variables/torch.py:
```diff
@@ -148,6 +148,37 @@ def _dynamo_overriden_transformers_eq(self, other):
+)
+
+
+def check_allowed_op(value):
```
I think we have more elegant way to handle this after #111622.
